File: Domain/Subscription/Model/Subscription.php
<?php

declare(strict_types=1);

namespace Domain\Subscription\Model;

use DateTime;
use Domain\Incident\Model\Incident;
use Domain\User\Model\User;

class Subscription
{
    private string $id;
    private NotificationChannel $channel;
    private DateTime $subscribedAt;
    private User $user;
    private Incident $incident;
    private NotificationCollection $notifications;

    public function __construct(string $id, NotificationChannel $channel, DateTime $subscribedAt, User $user, Incident $incident)
    {
        $this->id = $id;
        $this->channel = $channel;
        $this->subscribedAt = $subscribedAt;
        $this->user = $user;
        $this->incident = $incident;
        $this->notifications = new NotificationCollection();
    }

    public function getId(): string
    {
        return $this->id;
    }

    public function getChannel(): NotificationChannel
    {
        return $this->channel;
    }

    public function getSubscribedAt(): DateTime
    {
        return $this->subscribedAt;
    }

    public function getUser(): User
    {
        return $this->user;
    }

    public function getIncident(): Incident
    {
        return $this->incident;
    }

    public function getNotifications(): NotificationCollection
    {
        return $this->notifications;
    }
}

File: src/main/java/com/uptime/incident/domain/model/Subscription.java
package com.uptime.incident.domain.model;

import com.uptime.incident.domain.model.enums.NotificationChannel;

import java.time.Instant;

public class Subscription {
    private NotificationChannel notificationChannel;
    private Instant subscribedAt;

    public NotificationChannel getNotificationChannel() {
        return notificationChannel;
    }

    public void setNotificationChannel(NotificationChannel notificationChannel) {
        this.notificationChannel = notificationChannel;
    }

    public Instant getSubscribedAt() {
        return subscribedAt;
    }

    public void setSubscribedAt(Instant subscribedAt) {
        this.subscribedAt = subscribedAt;
    }
}
